app/routes/call.tsx

```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
+        display it with tabs like on the podcast page. [2023, 2022, 2021]
+      */}
+      <Grid className="mb-24 lg:mb-64">
+        <div className="flex flex-col col-span-full mb-6 lg:flex-row lg:justify-between lg:mb-12">
+          <H6
+            id="episodes"
+            as="h2"
+            className="flex flex-col col-span-full mb-10 lg:flex-row lg:mb-0"
+          >
+            <span>Calls with Kent C. Dodds</span>
+            &nbsp;
+            <span>{` — ${data.episodes.length} episodes`}</span>
+          </H6>
+
+          <button
+            className="text-primary inline-flex items-center text-lg font-medium"
+            onClick={() => setSortOrder(o => (o === 'asc' ? 'desc' : 'asc'))}
+          >
+            {sortOrder === 'asc' ? (
+              <>
+                Showing oldest first
+                <ChevronUpIcon className="ml-2 text-gray-400" />
+              </>
+            ) : (
+              <>
+                Showing newest first
+                <ChevronDownIcon className="ml-2 text-gray-400" />
+              </>
+            )}
+          </button>
+        </div>
+
+        <div className="col-span-full">
+          {sortedEpisodes.map((episode, idx, {length}) => {
+            const number =
+              sortOrder === 'asc' ? idx + 1 : Math.abs(length - idx)
+
+            return (
+              <div
+                className="border-b border-gray-200 dark:border-gray-600"
+                key={episode.slug}
+              >
+                <Link
+                  data-episode={episode.slug}
+                  key={episode.slug}
+                  to={activeSlug === episode.slug ? './' : `./${episode.slug}`}
+                >
+                  <Grid nested className="group relative py-10 lg:py-5">
+                    <div className="bg-secondary absolute -inset-px group-hover:block hidden -mx-6 rounded-lg" />
+                    <div className="relative flex-none col-span-1">
+                      <div className="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transform scale-0 group-hover:scale-100 transition">
+                        <div className="flex-none p-4 text-gray-800 bg-white rounded-full">
+                          <TriangleIcon size={12} />
+                        </div>
+                      </div>
+                      <img
+                        className="w-full rounded-lg object-cover"
+                        src={episode.imageUrl}
+                        alt={episode.title}
+                      />
+                    </div>
+                    <div className="text-primary relative flex flex-col col-span-3 md:col-span-7 lg:flex-row lg:col-span-11 lg:items-center lg:justify-between">
+                      <div className="mb-3 text-xl font-medium lg:mb-0">
+                        {/* TODO: make it three digits? How many calls do we expect? */}
```

Comment:
Yeah, let's be optimistic and do three digits 😅

Reply:
I've made them dynamic. :sunglasses: It now shows as many leading zeros as required, based on the number of episodes available. You're good until episode 9999. Probably that's the right time to add some windowing or pagination as well :sweat_smile:

Reply:
I like your idea of splitting things by year 👍 We'll see how things look if/when we get to a ton of episodes :)